Broken non regression tests

It seems like we have two broken tests at the moment: https://github.com/aramis-lab/clinica/actions/runs/13538941039/job/37904968563

```
=========================== short test summary info ============================
FAILED nonregression/iotools/test_run_utils.py::test_center_nifti - AssertionError: Following text files are different : 

 centered_nifti_list_20250225-151004.txt
FAILED nonregression/pipelines/anat/test_t1_linear.py::test_flair_linear - AssertionError: Following images do not meet the similarity criteria : 

sub-ADNI941S6580_ses-M000_space-MNI152NLin2009cSym_desc-Crop_res-1x1x1_FLAIR.nii.gz
sub-ADNI941S6580_ses-M000_space-MNI152NLin2009cSym_res-1x1x1_FLAIR.nii.gz
sub-02_ses-M000_run-123_space-MNI152NLin2009cSym_desc-Crop_res-1x1x1_FLAIR.nii.gz
sub-02_ses-M000_run-123_space-MNI152NLin2009cSym_res-1x1x1_FLAIR.nii.gz
= 2 failed, 12 passed, 1 skipped, 30 deselected, 134 warnings in 3546.52s (0:59:06) =
```

@AliceJoubert I think it is due to the recent work done in #1438 #1439 and #1441

I haven't looked in details but I think there are two distinct issues: one with the text file comparison and one with the nifti comparison (where it seems like the function isn't comparing the right images).

Do you think you can take a closer look ? (I you're busy with other things I can try to find some time to look into it).

--- a/test/nonregression/pipelines/anat/test_t1_linear.py
+++ b/test/nonregression/pipelines/anat/test_t1_linear.py
@@ -59,10 +59,10 @@ def run_flair_linear(
         bids_directory=fspath(input_dir / "bids"),
         caps_directory=fspath(out_caps),
         base_dir=fspath(working_dir),
-        parameters={"uncropped_image": False},
+        parameters={"uncropped_image": False, "random_seed": 1},
         name="flair-linear",
     )
     pipeline.run(plugin="MultiProc", plugin_args={"n_procs": 4}, bypass_check=True)
 
     compare_folders(out_caps, ref_caps, output_dir)
-    compare_niftis(out_caps, ref_caps, threshold=0.8)
+    compare_niftis(out_caps, ref_caps)

--- a/test/nonregression/pipelines/anat/test_t1_linear.py
+++ b/test/nonregression/pipelines/anat/test_t1_linear.py
@@ -65,4 +65,4 @@ def run_flair_linear(
     pipeline.run(plugin="MultiProc", plugin_args={"n_procs": 4}, bypass_check=True)
 
     compare_folders(out_caps, ref_caps, output_dir)
-    compare_niftis(out_caps, ref_caps)
+    compare_niftis(out_caps, ref_caps, threshold=0.8)

